web/src/pages/playground.tsx

```diff
@@ -207,33 +239,50 @@ const ChatSettings = ({ settings, updateSettings }: ChatSettingsParams) => {
       </select>
 
       <SectionHeader text="Token options" />
-      <NumberInput field="numTokens" label="Tokens" min="1" />
+      <NumberInput
+        value={settings.numTokens}
+        field="numTokens"
+        label="Tokens"
+        min="1"
+        max={MODELS[settings.completions as keyof typeof MODELS].numTokens}
+        updater={updateNum("numTokens")}
+      />
       <NumberInput
         field="tokensBuffer"
+        value={settings.tokensBuffer}
         label="Number of tokens to leave as a buffer when calculating remaining tokens"
         min="0"
-        max={settings.tokensBuffer}
+        max={settings.numTokens}
+        updater={updateNum("tokensBuffer")}
       />
 
       <SectionHeader text="Prompt options" />
       <NumberInput
+        value={settings.topKBlocks}
         field="topKBlocks"
```

Comment:
it there a point of `field` when passing value and updater manually?

Reply:
it keeps the label and styleing in one place